src/File/FileInformation.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\File;
+
+use Laminas\Filter\Exception\RuntimeException;
+use Psr\Http\Message\UploadedFileInterface;
+
+use function assert;
+use function basename;
+use function file_exists;
+use function finfo_open;
+use function is_array;
+use function is_readable;
+use function is_string;
+
+use const FILEINFO_MIME_TYPE;
+
+final class FileInformation
```

Comment:
General purpose file information utility. This would have been marked as `@internal`, but we use it to match unknown files to archive adapters in the interface. Marking it as `@internal` would have been a pain in the ass for anyone who wants to implement a different detection strategy